File: backend/src/main/java/me/sarismart/backend/Entity/Product.java
package me.sarismart.backend.Entity;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "products")
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String barcode;

    @Column(nullable = false)
    private String name;

    private String category;
    
    private String description;

    @Column(nullable = false)
    private double price;

    private int stock;

    private int sold;
    
    private int reorderLevel;

    @ManyToOne
    @JoinColumn(name = "store_id", nullable = false)
    private Store store;
}

File: frontend_mobile/app/src/main/java/edu/cit/sarismart/features/user/tabs/stores/data/models/Product.kt
package edu.cit.sarismart.features.user.tabs.stores.data.models

data class Product (
    val id: Long,
    val name: String,
    val category: String,
    val price: Double,
    val stock: Int,
    val description: String,
    val reorderLevel: Int,
    val sold: Int,
    val store: Store,
    val barcode: String
)